lms/tests/test_models.py

```diff
@@ -1,143 +1,251 @@
+
+    @staticmethod
+    def test_start_checking(
+            db_in_memory: SqliteDatabase,
             exercise: Exercise,
             student_user: User,
-    ):
-        first_solution = Solution.create_solution(exercise, student_user)
-        second_solution = Solution.create_solution(exercise, student_user)
-        assert second_solution.state == self.created_state
-        assert first_solution.refresh().state == self.old_solution_state
-
-        assert Solution.next_unchecked().id == second_solution.id
-        next_unchecked = Solution.next_unchecked_of(exercise.id)
-        assert next_unchecked.id == second_solution.id
-        assert next_unchecked.start_checking()
-        assert next_unchecked.refresh().state == self.in_checking_state
-
-        assert Solution.next_unchecked() is None
-        assert Solution.next_unchecked_of(exercise.id) is None
-        general_tasks.reset_solution_state_if_needed(
-            second_solution.id)
-        assert Solution.next_unchecked().id == second_solution.id
-        assert Solution.next_unchecked_of(exercise.id).id == second_solution.id
-
-    def test_next_exercise_with_cleanest_code(
-            self,
-            comment: Comment,
             staff_user: User,
     ):
-        student_user: User = conftest.create_student_user(index=1)
-        first_solution = comment.solution
-        comment_text = comment.comment
-        second_solution = Solution.create_solution(
-            comment.solution.exercise, student_user)
-
-        # comment exists on first solution - second one should be the first
-        assert Solution.next_unchecked().id == second_solution.id
-
-        # delete the comment should give us back the first solution
-        Comment.delete_by_id(comment.id)
-        assert Solution.next_unchecked().id == first_solution.id
-
-        # if second_solution has comments we should get first solution
-        Comment.create_comment(
-            commenter=staff_user,
-            line_number=1,
-            comment_text=comment_text,
-            solution=second_solution,
-            is_auto=False,
-        )
-        assert Solution.next_unchecked().id == first_solution.id
-
-        # both have comments - the first one should be the first solution
-        Comment.create_comment(
-            commenter=staff_user,
-            line_number=1,
-            comment_text=comment_text,
-            solution=first_solution,
-            is_auto=False,
-        )
-        assert Solution.next_unchecked().id == first_solution.id
+        student_user2 = conftest.create_student_user(index=1)
+        exercise2 = conftest.create_exercise(1)
+        solution1 = Solution.create_solution(exercise, student_user)
+        solution2 = Solution.create_solution(exercise2, student_user)
+        solution3 = Solution.create_solution(exercise, student_user2)
+
+        is_checking = solutions.start_checking(solution=None)
+        assert not is_checking
+
+        with no_celery:
```

Comment:
Why you need this?

Reply:
Because celery is eager, and it makes the 10 minutes to pass automatically, meaning the solution is not locked to a specific user.